pkg/provisioners/channel_util.go

```diff
@@ -169,6 +187,40 @@ func CreateVirtualService(ctx context.Context, client runtimeClient.Client, chan
 	return virtualService, nil
 }
 
+// checkExpectedLabels checks the presence of expected labels and its values and return true
+// if all labels are found.
+func checkExpectedLabels(actual, expected map[string]string) bool {
```

Comment:
I believe this is equivalent to `equality.Semantic.DeepDerivative(expected, actual)`.

Reply:
@Harwayne I get your point, I was just trying to be more specific in what I check and can control. I think usage `DeepDerivative` in this specific case is too generic, but I do not have a strong opinion about it.

Reply:
I have a slight preference towards using `DeepDerivative`, but it is weak considering how simple this function is. But, if we are going to keep it, please add unit tests for it.